### ISSUE TITLE
Input issues when writing a note on Android

### ISSUE THREAD

**Issue by mrjo118** [laurent22/joplin#9551]
### Operating system

Android

### Joplin version

2.13.10

### Current behaviour

https://github.com/laurent22/joplin/assets/6963004/ffa79796-a58b-4e70-a3a9-3dcb3727fdf6

When writing a note, sometimes random nearby text is duplicated at a nearby location, every time a character is entered (see attached video). This happened on an Android 10 phone using Joplin for Android 2.13.10, but I have experienced this issue a few times before in the previous version as well, which was Joplin 2.12.x. Once it starts happening, it happens continually until you exit out of the note, and it starts to happen frequently after that until you restart the phone.

I'm not sure how to consistently reproduce this, but these are the circumstances for which the bug happened:

- I edited a large existing note, which contained roughly 83,750 characters
- I searched for a phrase within the note which matched 10+ hits. After doing 'find next' several times for the phrase, I started editing some text near to the matched hit, writing new sentences in the middle of an existing paragraph
- The issue usually seems to happen when typing a period before an existing letter, or when typing a space after a period in the middle of an existing paragraph

### Expected behaviour

When writing a note, only the characters typed by the user should be entered, and they should be entered at the position of the cursor

**Comment by mrjo118** [comment]
Could this possibly be related to this issue?
https://github.com/jeremybarbet/react-native-portalize/issues/6

It mentions the same issue occurs in the react-native-paper project, which I see is used by Joplin

Also see: https://github.com/callstack/react-native-paper/issues/3097

**Comment by mrjo118** [comment]
Possibly this can be fixed by changing this line?:
https://github.com/laurent22/joplin/blob/880215acc436aae077f494dd9dd79539ad4693fb/packages/app-mobile/components/screens/Note.tsx#L1509

from:
value={note.body}
to:
defaultValue={note.body}

**Comment by personalizedrefrigerator** [comment]
> Possibly this can be fixed by changing this line?:
> 
> https://github.com/laurent22/joplin/blob/880215acc436aae077f494dd9dd79539ad4693fb/packages/app-mobile/components/screens/Note.tsx#L1509
> 
> from: value={note.body} to: defaultValue={note.body}

Good find! However, the video above suggests that the CodeMirror 6-based markdown editor is being used (and not the `TextInput` editor). 

Another possibility is that it's related to the Android keyboard being used and CodeMirror 6: Historically, [there have been input issues with Samsung Keyboard and autocorrect in CodeMirror](https://github.com/laurent22/joplin/issues/7861).

**Comment by mrjo118** [comment]
> > Possibly this can be fixed by changing this line?:
> > https://github.com/laurent22/joplin/blob/880215acc436aae077f494dd9dd79539ad4693fb/packages/app-mobile/components/screens/Note.tsx#L1509
> > 
> > from: value={note.body} to: defaultValue={note.body}
> 
> Good find! However, the video above suggests that the CodeMirror 6-based markdown editor is being used (and not the `TextInput` editor).
> 
> Another possibility is that it's related to the Android keyboard being used and CodeMirror 6: Historically, [there have been input issues with Samsung Keyboard and autocorrect in CodeMirror](https://github.com/laurent22/joplin/issues/7861).

I do have a Samsung Galaxy S9+ phone, however I don't use to Samsung keyboard as my input method and I turned off autocorrect on the keyboard because I never liked the feature. I use the Gboard app as my input method, however I also attempted changing the input method to Samsung keyboard while the issue occurred and the issue still persisted (and got even more buggy with the text body disappearing and reappearing as I typed). I tested that when reproducing the issue again after recording the video and thought that would rule out the keyboard being the issue

**Comment by mrjo118** [comment]
Ok, I've duplicated the particular note which I was getting the issue with, to avoid messing it up (having a discard changes feature would be nice :)).

I've been able to easily reproduce the issue repeatedly on that note by rapidly typing gibberish including lots of period, space, enter and backspace presses and moving the cursor from time to time. So it turns out if I use the Samsung keyboard from the beginning, instead of switching keyboard after the issue has occurred, I can't recreate the issue. So I tried updating my Gboard app (which happened to be very outdated) and after doing so the issue no longer occurs when using it as the input method
